src/slices-and-lifetimes/exercise.rs

```diff
@@ -62,6 +62,10 @@ struct Field<'a> {
     value: FieldValue<'a>,
 }
 
+trait ProtoMessage<'a>: Default + 'a {
```

Comment:
This is a pretty interesting form of lifetime annotation -- I think this will generate nice discussion!

Reply:
Yeah! Am I understanding it correctly that with

```rust
trait ProtoMessage<'_>: Default
```

you would end up with `'static` in the place of `'_`? I believe I've run into this situation before...

Reply:
That's my understanding. Without the lifetime annotation, you could have:
```rust
fn f<M: ProtoMessage>(m: M) {
  ALL_MESSAGES_EVER_SEEN.push(m)
}
```
and that has to be check-able without reference to the concrete type of M.